### ISSUE TITLE
[Bug] Edit Connection doesn't work for RECENTLY DUPLICATED CONNECTIONS / NEW CONNECTIONS  (PR attached)

### ISSUE THREAD

**Issue by synle** [sqlectron/sqlectron-gui#409]
**Edit Button DOES NOT for RECENTLY DUPLICATED / NEW CONNECTIONS**

Steps to repro
1. duplicate an existing connection / or create a brand new connection
2. try to edit it (notice it's not working)...

Connect button works, but the edit button doesn't work.

I'll check to see if I can send in a PR.

**Comment by synle** [comment]
Look like it could be a logic issue in the duplicate code.

After duplicate, the number of server is still showing up as 3 (instead of 4) like in this example...
![screen shot 2017-12-22 at 6 56 10 am](https://user-images.githubusercontent.com/3792401/34302360-4c0f6ac0-e6e5-11e7-9912-a5f7a39aee76.png)

**Comment by synle** [comment]
After some diggings, I found that the routine we use to find the servers is not correct.

We looked at at config (for some reasons it's out of sync)
```
const server = getState().config.data.servers.find(srv => srv.id === id);
```

Whereas we should use this value instead...
```
const server = getState().servers.items.find(srv => srv.id === id);
```

**Comment by synle** [comment]
Created a PR for this, PR link is here https://github.com/sqlectron/sqlectron-gui/pull/410

**Comment by synle** [comment]
Look like this issue also appears in RECENTLY CREATED CONNECTIONS.

**Comment by synle** [comment]
The fix should be pushed already in #410 .

What I did is to reload the config after save and after duplication

**Comment by synle** [comment]
PR merged, closing out
